### ISSUE TITLE
Move data binding (ItemsSource support) out of core series and DataPoint

### ISSUE THREAD

**Issue by tibel** [oxyplot/oxyplot#475]
The data binding feature is (mostly) used when defining a plot in XAML only.
But the binding logic adds much complexity to these classes (series and datapoint).

I think it would be better to move the data binding logic into some adapter classes.
Thus the core model will become easier to understand and optimize.
Also the whole databinding logic is in one place and can be optimized separately.

**Comment by objorke** [comment]
Good idea, I agree this could make the core implementation less complex.
